packages/middleware-token/src/tokenMiddleware.ts

```diff
@@ -21,12 +23,22 @@ export const tokenMiddleware =
   <Input extends object, Output extends object>(
     options: TokenResolvedConfig
   ): FinalizeRequestMiddleware<Input, Output> =>
-  (next: FinalizeHandler<Input, Output>): FinalizeHandler<Input, Output> =>
+  (next: FinalizeHandler<Input, Output>, context: HandlerExecutionContext): FinalizeHandler<Input, Output> =>
   async (args: FinalizeHandlerArguments<Input>): Promise<FinalizeHandlerOutput<Output>> => {
-    if (!HttpRequest.isInstance(args.request)) return next(args);
+    if (!HttpRequest.isInstance(args.request) || context.currentAuthConfig) return next(args);
 
-    const token = await options.token();
-    args.request.headers["Authorization"] = `Bearer ${token.token}`;
+    const token = options.token && (await options.token());
+    if (token?.token) {
+      const authConfig: HttpHeaderAuthDefinition = {
+        in: "header",
+        name: "Authorization",
+        scheme: "Bearer",
+      };
+      context.currentAuthConfig = authConfig;
+      args.request.headers[authConfig.name] = `${authConfig.scheme} ${token.token}`;
+    } else {
+      context.currentAuthConfig = null;
```

Comment:
add this field explicitly to the `HandlerExecutionContext` interface, so that static analysis / IDEs can pick up all the usages of it.